ksqldb-engine/src/main/java/io/confluent/ksql/KsqlExecutionContext.java

```diff
@@ -125,6 +129,16 @@ TransientQueryMetadata executeQuery(
       ConfiguredStatement<Query> statement
   );
 
+  PullQueryResult executePullQuery(
+      ServiceContext serviceContext,
+      RoutingFilterFactory routingFilterFactory,
+      ConfiguredStatement<Query> statement,
+      Map<String, Object> requestProperties,
+      Optional<Boolean> isInternalRequest,
```

Comment:
I am not sure I understand why it is clunky but @AlanConfluent wrote this code: Alan do you have any background information here of why this needs to be `Optional`? Can it be replaced with boolean? I see in `ServerVerticle:isInternalRequest` that it could just as easily return boolean (true if is internal, false otherwise), right?
```
  private static Optional<Boolean> isInternalRequest(final RoutingContext routingContext) {
    return Optional.ofNullable(routingContext.get(CONTEXT_DATA_IS_INTERNAL));
  }
```

Reply:
In some areas of code, it was necessary to see whether the internal interface was enabled at all.  Effectively, it's a 3 state flag: Internal interface not in use at all, internal interface in use and this request is internal, and internal interface in use and this request is not internal.  If you look at how it's being used in the code it's this:

```
final boolean isAlreadyForwarded = routingOptions.skipForwardRequest()
          // Trust the forward request option if isInternalRequest isn't available.
          && isInternalRequest.orElse(true);
```

So, if the interface isn't in use, the value defaults to true, not false.  If you want to do this at the calling layer, that's fine.  Also, if this three state thing isn's clear, it could be wrapped in a class that makes it clearer.

Reply:
that makes sense @AlanConfluent  thanks for the clarification, I think it would be nice to wrap it in an enum representing these three states (or in a class like you suggest) but this can be done in a follow-up (or now, if you're eager!)